src/ape/managers/chain.py

```diff
@@ -622,15 +624,24 @@ def __getitem_str(self, account_or_hash: str) -> Union[AccountHistory, ReceiptAP
             return self._get_account_history(address)
         except Exception:
             # Use Transaction hash
-            receipt = self._hash_to_receipt_map.get(account_or_hash)
-            if not receipt:
-                # TODO: Replace with query manager once supports receipts
-                #  instead of transactions.
-                # TODO: Add timeout = 0 once in API method to not wait for txns
-                receipt = self.provider.get_receipt(account_or_hash)
-                self.append(receipt)
+            try:
+                return self._get_receipt(account_or_hash)
+            except Exception:
+                pass
+
+            # If we get here, we failed to get an account or receipt.
+            # Raise top-level exception.
+            raise
 
-            return receipt
+    def _get_receipt(self, txn_hash: str) -> ReceiptAPI:
+        receipt = self._hash_to_receipt_map.get(txn_hash)
+        if not receipt:
+            # TODO: Replace with query manager once supports receipts
```

Comment:
Technically it supports receipts right now, it's just misnamed as transactions

Reply:
what is the best way to query a receipt by its known hash?